New jinja2 pinning makes it impossible to build our codebase

This isn't a bug in the codebase of locust per se, but rather a [new jinja2 pinning here](https://github.com/locustio/locust/commit/b85fbbfbac5c61db972c1727a69ad60fb93a91ca) which makes it impossible to use locust in our codebase in which a different library requires a version of jinja2 higher than 3.1.0.

Is there a way that we can get rid of this pinning in locust? Or is there any known workaround?

For what it's worth, unpinning Jinja2 to use version 3.1.1 in GNU Guix on Python 3.9.9 and Locust 2.8.6, I had only one new test failure, which look benign and unrelated:
```
=================================== FAILURES ===================================
________________ TestMasterWorkerRunners.test_distributed_shape ________________

self = <locust.test.test_runners.TestMasterWorkerRunners testMethod=test_distributed_shape>

    def test_distributed_shape(self):
        """
        Full integration test that starts both a MasterRunner and three WorkerRunner instances
        and tests a basic LoadTestShape with scaling up and down users
        """
    
        class TestUser(User):
            @task
            def my_task(self):
                pass
    
        class TestShape(LoadTestShape):
            def tick(self):
                run_time = self.get_run_time()
                if run_time < 2:
                    return 9, 9
                elif run_time < 4:
                    return 21, 21
                elif run_time < 6:
                    return 3, 21
                else:
                    return None
    
        with mock.patch("locust.runners.WORKER_REPORT_INTERVAL", new=0.3):
            test_shape = TestShape()
            master_env = Environment(user_classes=[TestUser], shape_class=test_shape)
            master_env.shape_class.reset_time()
            master = master_env.create_master_runner("*", 0)
    
            workers = []
            for i in range(3):
                worker_env = Environment(user_classes=[TestUser])
                worker = worker_env.create_worker_runner("127.0.0.1", master.server.port)
                workers.append(worker)
    
            # Give workers time to connect
            sleep(0.1)
    
            # Start a shape test
            master.start_shape()
            sleep(1)
    
            # Ensure workers have connected and started the correct amount of users
            for worker in workers:
                self.assertEqual(3, worker.user_count, "Shape test has not reached stage 1")
                self.assertEqual(
                    9, test_shape.get_current_user_count(), "Shape is not seeing stage 1 runner user count correctly"
                )
            self.assertDictEqual(master.reported_user_classes_count, {"TestUser": 9})
    
            # Ensure new stage with more users has been reached
            sleep(2)
            for worker in workers:
                self.assertEqual(7, worker.user_count, "Shape test has not reached stage 2")
                self.assertEqual(
                    21, test_shape.get_current_user_count(), "Shape is not seeing stage 2 runner user count correctly"
                )
            self.assertDictEqual(master.reported_user_classes_count, {"TestUser": 21})
    
            # Ensure new stage with less users has been reached
            sleep(2)
            for worker in workers:
                self.assertEqual(1, worker.user_count, "Shape test has not reached stage 3")
                self.assertEqual(
                    3, test_shape.get_current_user_count(), "Shape is not seeing stage 3 runner user count correctly"
                )
            self.assertDictEqual(master.reported_user_classes_count, {"TestUser": 3})
    
            # Ensure test stops at the end
            sleep(2)
            for worker in workers:
                self.assertEqual(0, worker.user_count, "Shape test has not stopped")
                self.assertEqual(
                    0, test_shape.get_current_user_count(), "Shape is not seeing stopped runner user count correctly"
                )
            self.assertDictEqual(master.reported_user_classes_count, {"TestUser": 0})
    
>           self.assertEqual("stopped", master.state)
E           AssertionError: 'stopped' != 'stopping'
E           - stopped
E           + stopping

locust/test/test_runners.py:1121: AssertionError
[...]
=============================== warnings summary ===============================
../../../gnu/store/l4nsxlz4hlxabj97vaylj1mwlvcwm4sy-python-pyquery-1.2.17/lib/python3.9/site-packages/pyquery/pyquery.py:50: 15 warnings
  /gnu/store/l4nsxlz4hlxabj97vaylj1mwlvcwm4sy-python-pyquery-1.2.17/lib/python3.9/site-packages/pyquery/pyquery.py:50: DeprecationWarning: inspect.getargspec() is deprecated since Python 3.0, use inspect.signature() or inspect.getfullargspec()
    name, inspect.getargspec(func).defaults)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED locust/test/test_runners.py::TestMasterWorkerRunners::test_distributed_shape
= 1 failed, 434 passed, 1 skipped, 8 deselected, 15 warnings in 349.99s (0:05:49) =
error: in phase 'check': uncaught exception:
%exception #<&invoke-error program: "python" arguments: ("-m" "pytest" "locust" "-k" "not test_default_headless_spawn_options and not test_default_headless_spawn_options_with_shape and not test_headless_spawn_options_wo_run_time and not test_html_report_option and not test_web_options and not test_skip_logging and not test_custom_exit_code and not test_webserver") exit-status: 1 term-signal: #f stop-signal: #f>
```

--- a/setup.cfg
+++ b/setup.cfg
@@ -48,7 +48,6 @@ install_requires =
     Flask-Cors >=3.0.10
     roundrobin >=0.0.2
     typing-extensions >=3.7.4.3
-    Jinja2 <3.1.0
     pywin32;platform_system=='Windows'
 
 [options.packages.find]

